shopinvader_api_address/routers/address.py

```diff
@@ -0,0 +1,323 @@
+from typing import Annotated
+
+from fastapi import APIRouter, Depends
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.extendable_fastapi.schemas import PagedCollection
+from odoo.addons.fastapi.dependencies import (
+    authenticated_partner,
+    authenticated_partner_env,
+    paging,
+)
+from odoo.addons.fastapi.schemas import Paging
+from odoo.addons.fastapi.utils import FilteredDomainAdapter
+from odoo.addons.shopinvader_schema_address.schemas import (
+    DeliveryAddress,
+    InvoicingAddress,
+)
+
+from ..schemas import (
+    DeliveryAddressCreate,
+    DeliveryAddressSearch,
+    DeliveryAddressUpdate,
+    InvoicingAddressCreate,
+    InvoicingAddressSearch,
+    InvoicingAddressUpdate,
+)
+
+# create a router
+address_router = APIRouter(tags=["addresses"])
+
+# --- Invoicing addresses ---
+
+
+@address_router.get("/addresses/invoicing")
+def get_invoicing_addresses(
+    params: Annotated[InvoicingAddressSearch, Depends()],
+    paging: Annotated[Paging, Depends(paging)],
+    env: Annotated[api.Environment, Depends(authenticated_partner_env)],
+    partner: Annotated[ResPartner, Depends(authenticated_partner)],
+) -> PagedCollection[InvoicingAddress]:
+    """
+    Get invoicing address of authenticated user
+    invoicing address corresponds to authenticated partner
+    """
+    count, addresses = env[
+        "shopinvader_api_address.invoicing_address_router.helper"
+    ]._search(partner, paging, params)
+    return PagedCollection[InvoicingAddress](
+        count=count, items=[InvoicingAddress.from_res_partner(rec) for rec in addresses]
+    )
```

Comment:
I think we had decided to not do paging nor search because there would never be tons of addresses for a given customer, and filtering could be easily done in the frontend. Just a remark en passant, nothing blocking.

Reply:
Indeed it make sens, but I would prefers to be consistent (a search always return a paginate).

I have a case where we plan to expose contact (it will done in shopinvader_api_contact, so we are going to expose a big list of contact ~200, so I will need paginatation. And it's look weird that on same object sometime we have pagination and sometime not)

Reply:
As I said, not blocking for me but I'd argue adding paging and search to small lists makes it more difficult for the front-end dev (unless they ignore it and assumes there will only ever be one page, but that is opening the door to bugs).

Reply:
I would also prefer to not do paging nor search on the addresses by default. (We could add a dedicated endpoint if necessary to some uses)